internal/migrations/migrator.go

```diff
@@ -307,6 +295,65 @@ func (m *Migrator) NotifyHeight(ctx context.Context, block *common.BlockContext,
 	return nil
 }
 
+// generateGenesisConfig generates the genesis config for the new chain based on the snapshot and the current
+// chain's consensus params. It saves the genesis file to the migrations/snapshots directory.
+// This function is called only once at the start height of the migration.
+// It is run asynchronously as we don't have access to the cometbft's state during replay.
+// Therefore we need to wait for the consensus params fn to be set before we can generate the genesis file.
+func (m *Migrator) generateGenesisConfig(ctx context.Context, cfgParams *common.NetworkParameters, snapshotHash []byte, genesisValidators []*chain.GenesisValidator, logger log.Logger) error {
+	// block until the m.consensusParamsFn is set
+	<-m.consensusParamsFnChan
+
+	// sanity check
+	if m.consensusParamsFn == nil {
+		return fmt.Errorf("consensus params getter not set")
```

Comment:
The error return is discarded by the caller presently, so we should either log these or make the caller unto a simple closure that only logs the outcome.